src/Symfony/Component/EventDispatcher/EventDispatcher.php

```diff
@@ -185,4 +220,22 @@ private function sortListeners($eventName)
         krsort($this->listeners[$eventName]);
         $this->sorted[$eventName] = call_user_func_array('array_merge', $this->listeners[$eventName]);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isPropagationStopped($event)
+    {
+        return in_array($event, $this->stoppedEvents);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stopPropagation($event)
+    {
+        if (!$this->isPropagationStopped($event)) {
+            $this->stoppedEvents[] = $event;
```

Comment:
making the EventDispatcher stateful is a nogo (this is leaking memory by keeping a reference to the object btw)